src/ncp/radio.cmake

```diff
@@ -47,8 +52,11 @@ target_link_libraries(openthread-rcp
     PUBLIC
         openthread-radio
     PRIVATE
-        openthread-hdlc
         openthread-spinel-rcp
         ot-config-radio
         ot-config
 )
+
+if(NOT OT_NCP_SPI)
+    target_link_libraries(openthread-rcp PRIVATE openthread-hdlc)
```

Comment:
@parag-silabs dont we need to link libraries for SPI?

Reply:
No, I was able to build and run rcp over spi.